Purchases/CodableExtensions/Store+Extensions.swift

```diff
@@ -21,20 +21,30 @@ extension Store: Decodable {
             throw decoder.throwValueNotFoundError(expectedType: Store.self, message: "Unable to extract a storeString")
         }
 
-        switch storeString {
-        case "app_store":
-            self = .appStore
-        case "mac_app_store":
-            self = .macAppStore
-        case "play_store":
-            self = .playStore
-        case "stripe":
-            self = .stripe
-        case "promotional":
-            self = .promotional
-        default:
+        guard let type = Self.mapping[storeString] else {
             throw CodableError.unexpectedValue(Store.self)
         }
+
+        self = type
+    }
+
+    private static let mapping: [String: Self] = Self.allCases
+        .filter { $0.name != nil }
+        .dictionaryWithKeys { $0.name! }
```

Comment:
Similar suggestion as https://github.com/RevenueCat/purchases-ios/pull/1236#discussion_r795604554

```swift
private static let mapping: [String: Self] = Self.allCases
    .reduce(into: [:]) { (result, store) in
        if let name = store.name { result[name] = store }
    }
```

Reply:
I started going down that path cause I thought it would be pretty elegant, but I'm feeling uneasy about needing to expose a `public` `protocol` for this implementation detail (because these are public types).

Reply:
Changed.